src/EditorFeatures/Core/Implementation/Workspaces/EditorTextFactoryService.cs

```diff
@@ -94,7 +94,7 @@ private unsafe SourceText CreateTextFromTemporaryStorage(ISupportDirectMemoryAcc
             using (var reader = new DirectMemoryAccessStreamReader(src + 1, streamLength / sizeof(char) - 1))
             {
                 var buffer = CreateTextBuffer(reader, cancellationToken);
-                return buffer.CurrentSnapshot.AsRoslynText(Encoding.Unicode);
+                return buffer.CurrentSnapshot.AsRoslynText(encoding ?? Encoding.Unicode);
```

Comment:
That's actually the crux of this fix. It doesn't modify the text buffer (by this point it's already been created). However it does honor the encoding that the caller specified (if any). The final encoding is no more than a property in the "SourceText". It doesn't affect the text buffer.